src/vue/init.js

```diff
@@ -0,0 +1,24 @@
+import "./logo.js";
+import Vue from "../../node_modules/vue/dist/vue.esm.browser.js";
+import vueI18n from "./i18n.js";
+
+const logoModel = {
+    CONFIGURATOR: CONFIGURATOR,
```

Comment:
If this object `logoModel` is only for the logo component, I think is better to pass only the minimum data that it will need. It makes it easier to understand and easier if we use the minimum variables and not complex objects that we don't use.  Maybe in a future we want more, but until then pass the complete `CONFIGURATOR` object seems too much.

So I think is better to pass only a `versionConfigurator` (or `configuratorVersion` if you apply my other suggestion). Something like:
```js
configuratorVersion: CONFIGURATOR.version,
```

BUT I've tried and it does not get attached to the variable and I think it will not react automatically when we change the CONFIGURATOR.version when loading. I'm very newbie to this. This can be achieved easy in some way o do we need to refresh the `configuratorVersion` in the same way we do for the other model variables?

Reply:
I will grab a look, I think it should be fine with smaller object. This was just the shortest path to get what I want 😆

Reply:
Ok have a try and I see what's happening. So Vue uses some js `getter/setter` magic to make objects reactive once you pass them to the `Vue` instance.

Since `version` is primitive it's passed by value so Vue has no idea that `CONFIGURATOR` has changed.

In general yes, components should know least amount possible. There are at least 2 ways approach this:
1. Keep it as a complex object. Once more or even all components are migrated more components will need access to `CONFIGURATOR`. Also it's important to remember `logo` component receives primitive strings https://github.com/betaflight/betaflight-configurator/blob/210cc162ee527d7f835d9d877f88ded903340c3d/src/vue/logo.js#L4. Only the root `#logo` https://github.com/betaflight/betaflight-configurator/blob/210cc162ee527d7f835d9d877f88ded903340c3d/src/main.html#L158  of it receives `CONFIGURATOR`, once more components are implemented single root will have more and more components which would need access to those values so this would make more sense.
2. other option would be to update the model in a similar way to https://github.com/betaflight/betaflight-configurator/blob/210cc162ee527d7f835d9d877f88ded903340c3d/src/js/main.js#L747

There are other ways to achieve this of course. But I like the current one since it hooks up into current system quite easy.

Reply:
With that being said, I think it would be even better idea to pass some of the other global objects like `FC.CONFIG`. This way that would even remove this https://github.com/betaflight/betaflight-configurator/blob/210cc162ee527d7f835d9d877f88ded903340c3d/src/js/main.js#L745 function all together

Reply:
True, this is something that needs to be decided. Not a problem for now. Let's see if some experienced Vue developer goes here and gives his opinion.